File: combine.py
import os

input_directory = "./input"
output_directory = os.getcwd()
output_file = os.path.join(output_directory, "output by .py file.txt")

# Check if there are any files in input_directory
if not any(os.scandir(input_directory)):
    print("Error: No files found in the input directory.")
    if os.path.exists(output_file):
        os.remove(output_file)
    exit(1)

# Remove existing output file
if os.path.exists(output_file):
    os.remove(output_file)

# Flag to track whether it's the first file
first_file = True


# Function to process files in a directory
def process_files_in_directory(directory):
    global first_file

    for root, dirs, files in os.walk(directory):
        for file in files:
            file_path = os.path.join(root, file)
            relative_path = os.path.relpath(file_path, input_directory)

            # Add heading with file name
            if first_file:
                with open(output_file, "a") as output:
                    output.write(f"combined by .py file :)\n\n\n----- ----- {relative_path} ----- -----\n")
                first_file = False
            else:
                with open(output_file, "a") as output:
                    output.write(f"\n\n----- ----- {relative_path} ----- -----\n")

            # Concatenate the content of the file
            with open(file_path, "r") as input_file:
                content = input_file.read()
                with open(output_file, "a") as output:
                    output.write(content)


# Process the input directory
process_files_in_directory(input_directory)

print("Combined files saved to", output_file)

File: combine.js
const fs = require("fs");
const path = require("path");

let firstFile = true;
const inputDirectory = "./input";
const outputDirectory = process.cwd();
const outputFile = path.join(outputDirectory, "output by .js file.txt");

function processFilesInDirectory(directory) {
    fs.readdir(directory, { withFileTypes: true }, (err, entries) => {
        if (err) {
            console.error(`Error reading directory: ${err.message}`);
            return;
        }

        entries.forEach((entry) => {
            const filePath = path.join(directory, entry.name);
            const relativePath = path.relative(inputDirectory, filePath);

            if (entry.isFile()) {
                // Add heading with file name
                if (firstFile) {
                    fs.appendFileSync(outputFile, `combined by .js file :)\n\n\n----- ----- ${relativePath} ----- -----\n`);
                    firstFile = false;
                } else {
                    fs.appendFileSync(outputFile, `\n\n----- ----- ${relativePath} ----- -----\n`);
                }

                // Concatenate the content of the file
                const content = fs.readFileSync(filePath, "utf8");
                fs.appendFileSync(outputFile, content);
            } else if (entry.isDirectory()) {
                processFilesInDirectory(filePath);
            }
        });
    });
}

// Ensure the output directory exists
if (!fs.existsSync(outputDirectory)) {
    fs.mkdirSync(outputDirectory);
}

// Check if the input directory is empty
if (fs.readdirSync(inputDirectory).length === 0) {
    console.error("Error: No files found in the input directory.");
    if (fs.existsSync(outputFile)) {
        fs.unlinkSync(outputFile);
    }
    process.exit(1);
}

// Remove the output file if it exists
if (fs.existsSync(outputFile)) {
    fs.unlinkSync(outputFile);
}

// Start processing files
processFilesInDirectory(inputDirectory);

console.log("Combined files saved to", outputFile);
